include/seqan3/search/algorithm/search.hpp

```diff
@@ -95,18 +108,31 @@ inline auto search(index_t const & index, queries_t && queries, config_t const &
     }
 }
 
-// DOC: insertion/deletion are with resp. to the query. i.e. an insertion is the insertion of a base into the query
-// that does not occur in the text at the position
-//!\brief \todo Document!
-template <typename index_t, typename queries_t>
+/*!\brief Search a query or a range of queries in an index.
+ *        It will not allow for any errors and will output all matches as positions in the text.
+ * \param[in] index String index to be searched. Must model seqan3::fm_index_concept.
+ * \param[in] queries A single query or a range of queries. A query must model std::ranges::RandomAccessRange,
+                      a range of queries must model std::ranges::SizedRange.
+ *
+ * ### Complexity
+ *
+ * Exponential in the numbers of errors.
+ *
+ * ### Exceptions
+ *
+ * Basic exception guarantee.
+ */
+template <typename queries_t>
 //!\cond
     requires std::ranges::RandomAccessRange<queries_t> ||
-             (std::ranges::ForwardRange<queries_t> && std::ranges::RandomAccessRange<value_type_t<queries_t>>)
+             (std::ranges::SizedRange<queries_t> && std::ranges::RandomAccessRange<value_type_t<queries_t>>)
```

Comment:
SizedRange does not imply ForwardRange so you cannot jut replace this. Please find out what your actual requirements are and specifiy those here. You need SizedRange only if you ever call size on the range. You need to random-access only if you call `[]`. If you need both, you need to require both :wink: If you just do a range-based for-loop a ForwardRange is usually sufficient.